Create v3_5 upgrade playbooks

mtnbikenc opened the pull request.

mtnbikenc commented:
@dgoodwin Could you give this a quick review just to make sure I'm covering everything?  The v3_3 playbooks will be removed in a separate PR.

dgoodwin commented on playbooks/common/openshift-cluster/upgrades/v3_5/master_config_upgrade.yml:
```diff
@@ -0,0 +1,16 @@
+- modify_yaml:
+    dest: "{{ openshift.common.config_base}}/master/master-config.yaml"
+    yaml_key: 'kubernetesMasterConfig.admissionConfig'
+    yaml_value:
```
Is this just the 3.4 hook file? You can remove this file and both references to it as a hook above, then re-introduce when/if 3.5 specific steps are required. Because we require you to be on 3.4 already to run this we know these changes have already been applied.

mtnbikenc commented:
Verified 3.4 -> 3.5 upgrade.

mtnbikenc commented:
aos-ci-test

detiber commented on roles/openshift_facts/library/openshift_facts.py:
```diff
@@ -915,7 +916,9 @@ def set_version_facts_if_unset(facts):
         facts['common']['version_gte_3_5_or_1_5'] = version_gte_3_5_or_1_5
         facts['common']['version_gte_3_6_or_1_6'] = version_gte_3_6_or_1_6
 
-        if version_gte_3_4_or_1_4:
```
Not really related to this line, but shouldn't version_gte_3_4_or_1_4 and version_gte_3_5_or_1_5 both default to true as well?

mtnbikenc replied:
I've updated the version defaults as you mentioned.  Where would I go to find what needed to be done for the new default predicates/priorities for the 3.5/1.5 release?

detiber replied:
You can find them in the defaultPredicates and defaultPriorities methods here: https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

mtnbikenc commented:
@detiber Predicates reordered according to Origin source and tests updated.  Priorities added for 1.5.

mtnbikenc commented:
aos-ci-test

mtnbikenc commented:
@dgoodwin could you follow up on your review?

mtnbikenc commented:
aos-ci-test

detiber commented on roles/openshift_master_facts/lookup_plugins/openshift_master_facts_default_predicates.py:
```diff
@@ -52,32 +52,34 @@ def run(self, terms, variables=None, regions_enabled=True, short_version=None,
             predicates.append({'name': 'PodFitsResources'})
-
-        # applies to all known versions
-        predicates.append({'name': 'NoDiskConflict'})
```
Looks like you are re-ordering the predicates for all versions, instead of just the latest version...

mtnbikenc commented:
@detiber Reverted predicates lookup and refactored to separate by version to allow reordering.  Reverted predicates tests and added test for 1.5 predicates.

mtnbikenc commented:
aos-ci-test

detiber commented on roles/openshift_master_facts/lookup_plugins/openshift_master_facts_default_priorities.py:
```diff
@@ -70,6 +70,12 @@ def run(self, terms, variables=None, zones_enabled=True, short_version=None,
+
+        if short_version not in ['1.1', '1.2', '1.3', '1.4']:
+            priorities.append({'name': 'BalancedResourceAllocation', 'weight': 1})
+
```
Should the tests for the priorities lookup plugin be updated for these changes as well?

sdodson commented:
aos-ci-test

mtnbikenc commented:
@sdodson @detiber I had to add a check so that we didn't change the firewall service across an upgrade boundary.  I just completed a test upgrade from release-1.4 successfully.

sdodson commented:
aos-ci-test

sdodson commented:
@detiber any remaining concerns?
